Add links to features section in README.md & features.md

shonamckenzie opened the pull request:
<!-- Provide a general summary of your changes in the title above -->

<!-- If this pull request is to check for a new a11y feature or modify an existing a11y feature check, please label it as `a11y feature` -->
<!-- If this pull request is to enhance anything else in the project (I.e. linting, dependencies, README, architecture, etc), please label it as `project enhancement` -->

## Types of changes
<!-- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] Bug fix (a non-breaking change which fixes an issue)

- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)


## Description
<!-- Describe your changes in clear detail -->
1. Created new references.md which contains features section and links to W3C resource
2. Updated README.md to remove individual features and add link to references.md
3. Updated CONTRIBUTING.md to reflect new features section location for pull request process

<!-- Why is this change required? What problem does it solve? -->
Added as validity of the features we have and why this project has them.

<!-- If it resolves an open issue, please link to the issue here. For example "Resolves: #1337" -->
Resolves #17 

## Link(s)
<!-- Please provide any relevant links used in your investigation in this work. -->
https://www.w3.org/TR/UNDERSTANDING-WCAG20/

## Checklist:
<!-- Put an `x` in all the boxes that apply. -->
<!-- Please do not submit the PR for review until most of the boxes are completed. -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] I have thoroughly read the [CONTRIBUTING](../CONTRIBUTING.md) guidelines.
- [x] I understand my pull request will be thoroughly reviewed at high detail.
- [x] I understand the work in my pull request will **only** be available in the next version release of Checka11y.css and not in the current version release.
- [x] I confirm the work in this pull request is valid according to my findings and is not something for anything personal.
- [x] I have updated the [README](../README.md) where and if applicable (still put an `x` if you have considered this but thought there was nothing to add or modify).
- [x] I have added myself to the `contributors` section in `package.json` (still put an `x` if you have considered this but decided not to add yourself).
- [x] I have checked I have **not** committed any **accidental** files.
- [x] I have tested all the main modern browsers (I.e. Chrome, Firefox, Edge, Safari - please leave this unchecked if there were any browsers listed you could not test and list them in the [help](#help) section with details why you couldn't test that browser)
- [x] All new and existing a11y checks still work correctly (compare your local `test/index.html` to the `test/index.html` in the `master` branch).

## Help
<!-- Please provide any details here that you require any further help or assistance with. -->
<!-- E.g. I could not test in Safari because I do not have access to an Apple device. -->

jackdomleo7 commented on README.md:
```diff
@@ -31,53 +31,7 @@ _Lightweight_ &bull; _Modern_ &bull; _Accessibile_ &bull; _Customisable_ &bull;
 
 Our features are always checked against the W3C accessibility guidelines.
 
-A list of every a11y concern Checka11y.css will check for and highlight:
```
Sorry if the issue description was too brief, I was thinking we could leave this section as it but add a note below this section saying, "For further details on the features listed above, please refer to the [feature references](./references.md)". It's really useful for a developer to come straight to this README and be able to get a really quick overview. 😊

shonamckenzie replied:
Ah sure, that's no problem - I probably should have clarified that with you but felt we may end up duplicating effort by keeping the README and references files updated with similar info

jackdomleo7 commented on references.md:
```diff
@@ -0,0 +1,51 @@
+  - They cannot be empty
+  - They are reachable by assistive technologies (no `aria-hidden`)
+
+Other features:
```
Let's leave this section out and just have this file focus on the a11y features? 🙂

jackdomleo7 commented on references.md:
```diff
@@ -0,0 +1,51 @@
+## Feature References
```
Really good research finding these links for secure validity! 💪

jackdomleo7 commented on references.md:
```diff
@@ -0,0 +1,51 @@
+
+A list of every a11y concern Checka11y.css will check for and highlight with links to relevant guidelines or techniques:
+
+- [Missing `alt` attributes on images](https://www.w3.org/TR/UNDERSTANDING-WCAG20/text-equiv-all.html)
```
Rather than wrapping the whole thing as a link, I think it would look cleaner seeing a "Read about this here" or similar link that sites next to the feature text.

jackdomleo7 reviewed (changes requested):
Thank you for doing this! Looking up those links is not an easy task, so well done. I believe all the links you found are valid, too.

Just a few changes, then it's ready to be merged 🙂

shonamckenzie commented:
Thanks @jackdomleo7 I've hopefully addressed all your comments - let me know if there is anything else needing changed.
I'm wondering whether we want to include the sub features in both the readme and references file, for example:
```
- Checks headings have meaningful content: Read about this here
   - They cannot be empty
   - They are reachable by assistive technologies (no aria-hidden
```
on either file do we maybe just want to have the feature listed itself rather than with the extra sub feature, for example in references.md just have:
`Checks headings have meaningful content: Read about this here`

maybe I'm complicating this too much..... 😃

jackdomleo7 commented:
@shonamckenzie I don't think you're complicating at all. You raise an excellent point. We want the developers to be able to easily match up a feature from the README to a feature in the references. What do you think is the best approach? 🙂

danspratling commented:
I would say that the `references.md` file should be the primary list of features we support (maybe the file should actually be renamed to `features.md`.

Then in readme we would just make note of a few most common issues found and then prominently feature the link to the other file. Readme shouldn't have a complete list, for a few reasons.

1. Readme only needs to convince people to use the software. We don't need to list every feature to do that
2. We want to keep readme short to make it easy to find necessary information. Most of the time, this isn't the features and a big feature list doesn't help with that.
3. Having a feature list in multiple placed doesn't help us when maintaining the project as we need to update 2 places.

I would say just list the 5 biggest issues and then link to the new file.

jackdomleo7 commented:
I agree with @danspratling. 😊 Could we go for a rename with `features.md` and in the README, state something like, "See a full list of features here."

Then I agree with the 5 biggest features (I think we could increase to 10 when the project gets larger). Maybe we could list:
1. Check for invalid HTML elements nested inside of `<a>` and `<button>`
2. Check for missing `alt` attributes on images
3. Check for media that is set to `autoplay`
4. Check `<li>` is the **only** direct child of `<ol>` and `<ul>`
5. Check for programmable-only `tabindex` attributes on invalid HTML elements

Unless you feel there are a better top 5?

shonamckenzie commented:
I also agree with this approach, think it looks a lot cleaner and you don't have to maintain the same info in multiple places.
PR is ready for review again ☺️

jackdomleo7 reviewed (approved):
Very nice PR! Thank you!
